### PR TITLE
bpo-29341: Added path.like object to docstring in posixmodule.c and posixmodule.c.h

### PR DESCRIPTION
**Pull request by BNMetrics**
https://bugs.python.org/issue29341

### CODE REVIEW

**Comment by BNMetrics** [comment]
All changes on the docstring have been verified and tested, and these `os` method do indeed take `PathLike` objects as the `path`.

File: Misc/NEWS.d/next/Core and Builtins/2018-10-25-20-53-32.bpo-29341.jH-AMF.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added path-like objects to docstrings of os methods that accept path
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2018-10-25-20-53-32.bpo-29341.jH-AMF.rst]: I find this a bit confusing. I would rewrite so it says:

```
Clarify in the docstrings of :mod:`os` methods that path-like objects are also accepted
as input parameters.
```

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -2542,7 +2542,7 @@ os_lstat_impl(PyObject *module, path_t *path, int dir_fd)
 os.access -> bool
 
     path: path_t
-        Path to be tested; can be string or bytes
+        Path to be tested; can be string, bytes, or path-like object
```
Comment (pablogsal) [inline Modules/posixmodule.c]: I think this needs an "a": `or **a** path-like object`? (The same for the rest of the methods)

**Comment by BNMetrics** [comment]
@pablogsal Thanks so much for the review. I have made the changes suggested now! 👍

File: Modules/posixmodule.c [hunk 3]
```diff
@@ -2542,7 +2542,7 @@ os_lstat_impl(PyObject *module, path_t *path, int dir_fd)
 os.access -> bool
 
     path: path_t
-        Path to be tested; can be string or bytes
```
Comment (pablogsal) [inline Modules/posixmodule.c:2545]: Could you add a `.` at the end of this sentence?  :)

`Path to be tested; can be string, bytes, or a path-like object.`

Comment (BNMetrics) [reply Modules/posixmodule.c:2545]: Sure! :)

File: Modules/posixmodule.c [hunk 4]
```diff
@@ -12836,7 +12836,7 @@ os.scandir
 
 Return an iterator of DirEntry objects for given path.
 
-path can be specified as either str, bytes or path-like object.  If path
+path can be specified as either str, bytes or a path-like object.  If path
```
Comment (pablogsal) [inline Modules/posixmodule.c]: For consistency with the use of the oxford comma (the comma before the "or") in other docstrings we should use it here as well:

```suggestion
path can be specified as either str, bytes, or a path-like object.  If path
```

Comment (pablogsal) [reply Modules/posixmodule.c]: (If you apply this suggestion using GitHub, remember to regenerate the clinic file)

**Review by pablogsal — APPROVED** [review]
Thank you very much @BNMetrics for the PR and for the patience going through the review! 🎉 

LGTM

**Comment by BNMetrics** [comment]
Thank you so much @pablogsal !

**Comment by pablogsal** [comment]
Yup, I think it was a bit naive to think that the automatic backport will work. This is because the clinic file needs to be correctly regenerated. @BNMetrics Do you want to do the manual backport PRs? If you don't want to I can do them myself :)

**Comment by pablogsal** [comment]
Ignore that, is a random failure

**Pull request merged**